pkg/snapstore/swift_snapstore.go

```diff
@@ -490,36 +492,51 @@ func (s *SwiftSnapStore) Delete(snap brtypes.Snapshot) error {
 	return result.Err
 }
 
-// SwiftSnapStoreHash calculates and returns the hash of openstack swift snapstore secret.
-func SwiftSnapStoreHash(config *brtypes.SnapstoreConfig) (string, error) {
-	if dir, isSet := os.LookupEnv(swiftCredentialFile); isSet {
-		swiftConfig, err := readSwiftCredentialDir(dir)
+// GetSwiftCredentialsLastModifiedTime returns the latest modification timestamp of the Swift credential file(s)
+func GetSwiftCredentialsLastModifiedTime() (time.Time, error) {
+	if dir, isSet := os.LookupEnv(swiftCredentialDirectory); isSet {
+		swiftTimeStamp, err := getSwiftBackupCredentialLatestModifiedTime(dir)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials from %v directory", dir)
+			return time.Time{}, fmt.Errorf("failed to get Swift credential timestamp from the directory %v with error: %v", dir, err)
 		}
-		return getSwiftHash(swiftConfig), nil
+		return swiftTimeStamp, nil
 	}
 
 	if filename, isSet := os.LookupEnv(swiftCredentialJSONFile); isSet {
-		swiftConfig, err := swiftCredentialsFromJSON(filename)
+		fileInfo, err := os.Stat(filename)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials using %v file", filename)
+			return time.Time{}, fmt.Errorf("failed to fetch file information of the Swift JSON credential file %v with error: %v", filename, err)
 		}
-		return getSwiftHash(swiftConfig), nil
+		return fileInfo.ModTime(), nil
 	}
 
-	return "", nil
+	return time.Time{}, fmt.Errorf("no environment variable set for the Swift credential file")
 }
 
-func getSwiftHash(config *swiftCredentials) string {
-	var data string
-	if config.AuthType == authTypeV3ApplicationCredential {
-		data = fmt.Sprintf("%s%s%s%s%s%s", config.AuthURL, config.TenantName, config.ApplicationCredentialID, config.DomainName, config.ApplicationCredentialName, config.ApplicationCredentialSecret)
+// getSwiftBackupCredentialLatestModifiedTime reads the modification timestamp of each Swift credential file and returns the latest
+func getSwiftBackupCredentialLatestModifiedTime(dirname string) (time.Time, error) {
+	_, err := os.Stat(dirname)
+	// the access credential directory has errored
+	if err != nil {
+		return time.Time{}, err
+	}
+	// credential files which are essential for creating the snapstore
+	var credentialFiles []string
+
+	// checking authType
+	if _, err := os.Stat(filepath.Join(dirname, "applicationCredentialID")); err != nil {
+		if _, err := os.Stat(filepath.Join(dirname, "password")); err != nil {
+			// both authTypes have errored
+			return time.Time{}, fmt.Errorf("not able to identify authentication type from the Swift credentials present")
+		}
+		// authTypePassword
+		credentialFiles = []string{"authURL", "tenantName", "domainName", "username", "password"}
```

Comment:
For checking the authType, you can re-use this function: 
https://github.com/gardener/etcd-backup-restore/blob/6a6e63fb970eec01b90d18bd05381f4130038c5a/pkg/snapstore/swift_snapstore.go#L540

Reply:
Addressed by commit [51ae353](https://github.com/gardener/etcd-backup-restore/pull/670/commits/51ae3533ad16c589fa8bf859ec52cb0da7b392c5)